Session cookie fix

kmorozov opened the pull request:
Constructor parameter for persisting session cookies for the SessionCookieJar and the FileCookieJar.
Ref: https://github.com/guzzle/guzzle/issues/1153

mtdowling commented on src/Cookie/FileCookieJar.php:
```diff
@@ -44,8 +49,10 @@ public function save($filename)
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
-                $json[] = $cookie->toArray();
+            if($cookie->getExpires() || $this->storeSessionCookies) {
```
What do you think about refactoring this duplication of if statements out into a static method on CookieJar? Maybe the method could be `shouldPersist(SetCookie $cookie, $allowSessionCookies)` or something similar?

kmorozov replied:
That seems reasonable. I'll make the change.

mtdowling commented:
Nice work!

Can you make sure to limit lines to 80 characters, ensure that braces are surrounded by spaces, and add tests? After that, I'll merge.

kmorozov commented:
I've added the tests and fixed the formatting. Do you have an Eclipse formatter? I've made the changes manually, but it would be easier if Eclipse did it for me :)

mtdowling commented on src/Cookie/CookieJar.php:
```diff
@@ -75,6 +76,26 @@ public static function getCookieValue($value)
+        SetCookie $cookie,
+        $allowSessionCookies = false
+    ) {
+    	if($cookie->getExpires() || $allowSessionCookies) {
```
Missing a space here between "if" and "("

mtdowling commented on src/Cookie/FileCookieJar.php:
```diff
@@ -44,7 +50,7 @@ public function save($filename)
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
+            if(CookieJar::shouldPersist($cookie, $this->storeSessionCookies)) {
```
Missing a space here between "if" and "("

mtdowling commented on src/Cookie/SessionCookieJar.php:
```diff
@@ -36,7 +43,7 @@ public function save()
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
+            if(CookieJar::shouldPersist($cookie, $this->storeSessionCookies)) {
```
Missing a space here between "if" and "("

mtdowling commented on tests/Cookie/FileCookieJarTest.php:
```diff
@@ -63,9 +63,19 @@ public function testPersistsToFileFile()
 
-        // Weeds out temporary and session cookies
-        $this->assertEquals(2, count($jar));
+        if($testSaveSessionCookie) {
```
Missing a space here between "if" and "("

mtdowling commented on tests/Cookie/SessionCookieJarTest.php:
```diff
@@ -67,9 +67,19 @@ public function testPersistsToSession()
 
-        // Weeds out temporary and session cookies
-        $this->assertEquals(2, count($jar));
+        if($testSaveSessionCookie) {
```
Missing a space here between "if" and "("

GrahamCampbell replied:
@mtdowling Maybe you should enable https://styleci.io/ on this repo so you don't have to review cs manually?

mtdowling commented on tests/Cookie/SessionCookieJarTest.php:
```diff
@@ -36,9 +36,9 @@ public function testLoadsFromSession()
     }
 
-    public function testPersistsToSession()
+    public function testPersistsToSession($testSaveSessionCookie = false)
```
This should have a `@dataProvider` that invokes the test. See https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.dataProvider

mtdowling commented on tests/Cookie/SessionCookieJarTest.php:
```diff
@@ -67,9 +67,19 @@ public function testPersistsToSession()
+
+    public function testPersistsToSessionWithSessionCookie()
+    {
+        $this->testPersistsToSession(true);
```
This should be removed and instead tested using a data provider. See https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.dataProvider

mtdowling commented:
Sweet! Thanks!
